articles/random-search/index.md

```diff
@@ -0,0 +1,221 @@
+
+![randomgrid](/engineering-education/random-search/randomgrid.png)
+
+*Random layout.*
+
+[Source](https://jmlr.csail.mit.edu/papers/volume13/bergstra12a/bergstra12a.pdf)
+
+#### RandomizedSearchCV
+
+We used the sci-kit learn (sklearn) library when implementing grid search, particularly GridSearchCV. From the same library, we shall use RandomizedSearchCV. Similar to GridSearchCV, it is meant to find the best parameters to improve a given model. A key difference is that it does not test all parameters. Instead, the search is done at random. As we shall note later, a practical difference between the two is that RandomizedSearchCV allows us to specify the number of parameter values we seek to test. This is through the use of ` n_iter. `
+
+#### Random Forest
+
+Since we shall use a random forest regressor during random search implementation, it is of value to introduce random forests. Random forests refer to an ensemble of untrained decision trees capable of both regression and classification tasks. They involve the use of bagging, which combines many models to give a generalized result. Learn more about bagging and ensemble learning as a whole from this [article](https://www.section.io/engineering-education/ensemble-learning/).
+We have explained why a random forest is a “forest” but not why it is considered random. The random nature of a random forest can be attributed to a couple of concepts. Firstly, the samples of training observations when building trees are taken randomly. Secondly, when it comes to the splitting of nodes, random subsets of features are used.
+
+##### Random forest parameters
+
+A random forest uses many parameters. Below are the main ones. It is essential to introduce them since we shall encounter all of them in our code later on.
+
+**n_estimators.** This parameter denotes the maximum number of trees in an ensemble/forest.
+
+**max_features.** This represents the maximum number of features taken into consideration when splitting a node.
+
+**max_depth.** max_depth represents the maximum number of levels that are allowed in each decision tree.
+
+**min_samples_split.** To cause a node to split, a minimum number of samples are required in a node. This minimum number of data points is what is represented by to as min_samples_split.
+
+**min_samples_leaf.** The minimum count of data points that can be stored in a leaf node.
+
+**bootstrap.** To sample data points, the bootstrap sampling method is used. Sampling may be carried out with or without replacement. Sampling with replacement can be described as when a sample is selected from a random population, then returned to the population.
```

Comment:
Does bootstrap carry out sampling with or without replacement?

Reply:
According to the material, it does with replacement if bootstrap = true, and without if = false.